sim/validator.py

```diff
@@ -30,14 +30,25 @@ def run(cmd, *args, **kwargs):
             cmd failed with exit code %i
           *****""" % (cmd, returncode))
 
-def simulate_genome(reference_path, output_path, num_snps=16000):
-    run([
-        "simuG.pl",
-        "-refseq", reference_path,
-        "-snp_count", str(num_snps),
-        "-prefix", output_path + "simulated"
-    ])
-
+def simulate_genome(predef_snp_path, reference_path, output_path, num_snps=16000, seed=1):
+    if predef_snp_path:
+        run([
+            "simuG.pl",
+            "-refseq", reference_path,
+            "-snp_vcf", predef_snp_path,
+            # below line tells simuG to also simulate predefined indels.
+            #"-indel_vcf", predef_snp_path, 
+            "-prefix", output_path + "simulated",
+            "-seed", str(seed)
+        ])
+    else:
+        run([
+            "simuG.pl",
+            "-refseq", reference_path,
+            "-snp_count", str(num_snps),
+            "-prefix", output_path + "simulated",
+            "-seed", str(seed)
+        ])
```

Comment:
This is a bit weird. `predef_snp_path` is a required argument, not possible for the `else` to be run unless you supply it `None` or an empty string

I think a better way to do this is have separate functions. They could be named something along the lines of `simulate_genome_from_vcf` and `simulate_genome_with_random_snps`

Reply:
You can also DRY up out the common parts of the command with a list of strings